packages/app/src/App.vue

```diff
@@ -11,6 +11,25 @@
   </router-view>
 </template>
 
+<script lang="ts" setup>
+import { onBeforeUnmount } from 'vue'
+import { gql, useMutation } from '@urql/vue'
+import { App_ShowLaunchpadDocument } from './generated/graphql'
+
+gql`
+mutation App_ShowLaunchpad {
+  showLaunchpadOnAppExit
+}
+`
+
+const showLaunchpad = useMutation(App_ShowLaunchpadDocument)
+
+window.addEventListener('beforeunload', () => {
```

Comment:
I tried doing this and some Vue lifecycle functions but they only worked sometimes. It seems like there is a race to for the network request to finish going through.

Reply:
I think I am missing some context; when is this supposed to be triggered? Eg, when is the `beforeunload` event called?